cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -342,16 +336,26 @@ static int _sc_cgroup_v2_init_bpf(sc_device_cgroup *self, int flags) {
         debug("bpffs mounted at /sys/fs/bpf");
     }
 
-    if (mkdir("/sys/fs/bpf/snap", 0700) < 0) {
-        if (errno != EEXIST) {
-            die("cannot create /sys/fs/bpf/snap directory");
+    /* TODO: open("/sys/fs/bpf") and then mkdirat?  */
```

Comment:
given that we are making a change to mkdir+chmod, shouldn't we paranoid and do switch to mkdirat/fchownat ?

Reply:
It's not a direct implication, but indeed we could. Can I do it in a follow-up though, since I already got a +1 from security?

Reply:
This sounds like a Good Thing - so I think it would be good to include here - I can re-review it quickly again if needed.

Reply:
Updated!